packages/js-script-writer/src/agent-functions/think.ts

```diff
@@ -1,14 +1,33 @@
-import { ResultOk } from "@polywrap/result";
-import { AgentFunction, AgentFunctionResult, BasicAgentMessage } from "@evo-ninja/agent-utils";
+import { Result, ResultOk } from "@polywrap/result";
+import { AgentFunction, AgentFunctionResult, FunctionCallMessage } from "@evo-ninja/agent-utils";
 import { AgentContext } from "../AgentContext";
 import { OTHER_EXECUTE_FUNCTION_OUTPUT } from "../prompts";
 
 const FN_NAME = "think";
-
 type FuncParameters = { 
   thoughts: string
 };
 
+const SUCCESS = (params: FuncParameters): AgentFunctionResult => ({
+  outputs: [
+    {
+      type: "success",
+      title: `Thinking...`,
+      content: 
+        `## Function Call:\n\`\`\`javascript\n${FN_NAME}\n\`\`\`\n` +
+        OTHER_EXECUTE_FUNCTION_OUTPUT(`I think: ${params.thoughts}.`)
+    }
+  ],
+  messages: [
+    new FunctionCallMessage(FN_NAME, params),
+    {
+      role: "system",
+      content: `## Function Call:\n\`\`\`javascript\n${FN_NAME}\n\`\`\`\n` +
```

Comment:
NIT: the content string could be deduplicated, and initialized above this return statement.

Reply:
In the next PR i'm planing to change the prompting so I didn't want to refactor something that I'll have to revert immediately